diff -Nuar a/lib/k3screenctrl/basic.sh b/lib/k3screenctrl/basic.sh
--- a/lib/k3screenctrl/basic.sh 2024-10-15 14:09:03.012286755 +0800
+++ b/lib/k3screenctrl/basic.sh 2024-10-27 02:40:55.431205205 +0800
@@ -5,8 +5,10 @@
 PRODUCT_NAME_FULL=$(cat /etc/board.json | jsonfilter -e "@.model.name")
 PRODUCT_NAME=${PRODUCT_NAME_FULL#* } # Remove first word to save space

-WAN_IFNAME=$(uci get network.wan.ifname)
+WAN_IFNAME=$(uci get network.wan.device)
 MAC_ADDR=$(ifconfig $WAN_IFNAME | grep -oE "([0-9A-Z]{2}:){5}[0-9A-Z]{2}")
+LAN_ADDR=$(uci get network.lan.ipaddr)
+WAN_ADDR=$(ifconfig wan | grep 'inet addr' | awk -F'[: ]+' '{print $4}')

 CPU_TEMP=$(($(cat /sys/class/thermal/thermal_zone0/temp) / 1000))

@@ -14,19 +16,23 @@
 #${LEDE_DEVICE_REVISION:0:2}
 FW_VERSION=${DISTRIB_REVISION:0:17}

-echo $PRODUCT_NAME
+#echo $PRODUCT_NAME

 if [ $(uci get k3screenctrl.@general[0].showmore) -eq 1 ]; then
-    echo U:$CPU_TEMP *C
-    used=`free | grep Mem | awk '{print$3}'`
+    RAM_free=`free | grep Mem | awk '{print$4}'`
+       used=`free | grep Mem | awk '{print$3}'`
     all=`free | grep Mem | awk '{print$2}'`
     LOAD=`uptime | awk -F "average:" '{print$2}' | awk -F "," '{print$1}'`
     UPTIME=`uptime | awk -F "," '{print$1}'|awk '{print"up " $3" " $4}'`
-    echo U:$LOAD R:$((100*$used/$all))%
-    echo $UPTIME
-    echo $DISTRIB_DESCRIPTION
+       CPU_used=`vmstat | tail -1 | awk '{ print 100 - $15 }'`         #需要安装PACKAGE_procps-ng包
+    echo $PRODUCT_NAME - $UPTIME
+       echo U:$CPU_TEMP*C $CPU_used% $LOAD
+       echo RU:$((100*$used/$all))% RF:$(($RAM_free/1024))M
+    echo L:$LAN_ADDR
+    echo W:$WAN_ADDR
 else
-    echo $HW_VERSION
+    echo $PRODUCT_NAME
+       echo $HW_VERSION
     echo $FW_VERSION
     echo $FW_VERSION
     echo $MAC_ADDR
